### PR TITLE
Weekly Calendar as a CSS Grid

### PR DESCRIPTION
**Pull request by jrjohnson**
This modernizes the weekly calendar to use better CSS and improved ember Octane patterns to pave the way for improved calendar functionality.

### CODE REVIEW

File: addon/components/weekly-calendar.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+export default class WeeklyCalendarComponent extends Component {
+  @service intl;
+  @service moment;
+
+  @action
+  scrollView(calendarElement, [earliestHour]) {
+    if (earliestHour === 24 || earliestHour < 2) {
+      return;
+    }
+    calendarElement.scrollTop = this[`hour${earliestHour - 2}`].offsetTop;
```
Comment (stopfstedt) [inline addon/components/weekly-calendar.js]: i don't get this - what is happening here with `this`?

Comment (jrjohnson) [reply addon/components/weekly-calendar.js]: That deserves a comment for sure, I'll add it.  

All of the hour elements are registering themselves here https://github.com/ilios/common/pull/1177/files?file-filters%5B%5D=.hbs#diff-5d032c49f2dc0c166fff5055177a5573R41. As hour1, hour2, etc. Scrolling to the actual hour didn't work though, for whatever reason I had to scroll back two hours. So this lines gets `offsetTop` from `this.hour7` (7 being two hours before the first session happens).

**Comment by stopfstedt** [comment]
upon further review, i'm finding more issues.

empty hours collapsing onto themselves. a min-height could fix that, i presume.

Selection_016

some overlap of events, compare old vs. new. the 9:40a event seems to have gone AWOL in the new calendar as well.

Selection_017
Selection_018

**Comment by dartajax** [comment]
It is not possible to scroll past 4:30pm.

image

**Comment by dartajax** [comment]
Header is out of alignment as well. See below.

image

**Comment by jrjohnson** [comment]
Can you add a mock-up of the full width showing how it should be aligned?

**Comment by jrjohnson** [comment]
I've believe I have addressed all of the above and this is ready for testing again.

**Comment by saschaben** [comment]
looks like we've lost the date identifiers for the day-of-week headers:

demo:
demo1

grids:
grids

**Comment by jrjohnson** [comment]
@saschaben / @dartajax ready for review again.

**Comment by dartajax** [comment]
Looks good to me - assigned to @stopfstedt for code review.

**Pull request merged**